fix(theming): light text on colored backgrounds should be opaque

Maistho opened the pull request:
According to [Material Design Guidelines](https://material.io/guidelines/style/color.html#color-usability), white text on colored backgrounds should be displayed at an opacity of 100%.

I also changed the naming of the variables to [dark/light]-primary-text instead of [black/white]-87-opacity, since it better reflects the names used in the guidelines.

This PR fixes #6236

Maistho commented:
I signed it!

jelbourn commented:
@josephperrott can you take a look to see how this compares to the spec / mdc-web?

josephperrott commented:
@jelbourn 
These changes match up with the values currently in the [spec](https://material.io/guidelines/style/color.html#color-usability) (under the heading `Text on backgrounds`)

MDC uses the same opacity in their [theming](https://github.com/material-components/material-components-web/blob/master/packages/mdc-theme/_variables.scss#L56)

@Maistho Can you rebase the PR?

Maistho commented:
@josephperrott I've rebased it!

josephperrott commented:
@Maistho Looks like there are some build errors now?

Maistho commented:
@josephperrott Sorry, I had apparently used `$dark-disabled` in some places and `$dark-disabled-text` in others.

Fixed it and rebased on master again.

jelbourn commented on src/lib/core/theming/_palette.scss:
```diff
@@ -6,12 +6,16 @@
+$dark-disabled-text: rgba(black, 0.38);
+$dark-dividers: rgba(black, 0.12);
+$dark-focused: rgba(black, 0.12);
+$light-primary-text: rgba(white, 1);
```
`rgba(white, 1)` can just be `white`

Maistho replied:
I felt like it was more clear/explicit that the opacity should be 100% if I typed it out, but I can change it if you prefer it.

jelbourn replied:
Yeah, I prefer to use the plain value when it is desired

Maistho replied:
Fixed and rebased

jelbourn commented:
Caretaker note: this PR will likely affect many screenshot tests and should be synced by itself. Reach out to teams to ensure that everything still looks correct.

Maistho commented:
Hi, what's the status on this PR? Anything I can do to help speed things up?

jelbourn commented:
Our merge process is just a bit behind at the moment; see https://github.com/angular/material2/blob/master/CODE_REVIEWS.md#how-prs-are-merged for more context.

This change is particularly involved because it involves updating many screenshot tests.

jelbourn commented:
@Maistho looks like there is a test failure on CI- can you rebase and see if it goes away?

jelbourn commented:
@Maistho can you also restore the removed scss variables and mark them as deprecated? Upon running an internal presubmit it turns out quite a large number of apps consume on them.

Maistho commented:
@jelbourn I've rebased it!

How do I add the deprecation notice? Not sure if you use [sassdoc](http://sassdoc.com/), if I should add it to the commit message, or if there's some other way.

I added sassdoc compatible comments for now.

jelbourn commented:
Looks good to me

andrewseguin commented:
@jelbourn Tests pass - do you consider this a patch or minor

jelbourn commented:
I would consider this acceptable for a patch because it only changes colors

andrewseguin commented:
@Maistho Thanks for your patience as we try and get this merged in. We're seeing some screenshot diffs get brought up that might point out something incorrect in this PR. 

Can you look at these screenshots and confirm the appearance of the disabled button? Looks like your change made it darker which in my eyes [doesn't match the spec](https://material.io/guidelines/components/buttons.html#buttons-raised-buttons).

**Spec**

![xdtf190pzkp](https://user-images.githubusercontent.com/22898577/34841120-0000c912-f6bc-11e7-9f07-51bcbfbc3955.png)

**Master (demo app, button page)**

![master](https://user-images.githubusercontent.com/22898577/34841077-dd76c2c0-f6bb-11e7-8bba-735cc0be2911.png)

**pr/4721 (demo app, button page)**

![new](https://user-images.githubusercontent.com/22898577/34841086-e4a088f6-f6bb-11e7-9ac5-197d5fb29ed7.png)

Maistho commented:
@andrewseguin Yup, I made a mistake there. Looks like the disabled buttons are using the text color for background. I'll take a look!

Maistho commented:
@andrewseguin rebased and pushed a fixed version
